src/modules/esl-carousel/core/esl-carousel.renderer.ts

```diff
@@ -127,13 +130,25 @@ export abstract class ESLCarouselRenderer implements ESLCarouselConfig {
 
   /** Sets active slides from passed index **/
   public setActive(from: number): void {
-    const count = Math.min(this.count, this.size);
     this.$carousel.$slides.forEach((el) => el.active = false);
+    const count = Math.min(this.count, this.size);
     for (let i = 0; i < count; i++) {
       this.$carousel.slideAt(from + i).active = true;
     }
   }
 
+  public setPreActive(from: number): void {
+    this.clearPreActive();
+    const count = Math.min(this.count, this.size);
+    for (let i = 0; i < count; i++) {
+      this.$carousel.slideAt(from + i).preActive = true;
+    }
```

Comment:
```suggestion
    for (let i = from; i < from + count; i++) {
      this.$carousel.slideAt(i).preActive = true;
    }
```
I guess that would be clearer at a glance.